tools/test/build_api/build_api_test.py

```diff
@@ -238,5 +240,34 @@ def test_build_library_no_app_config(self, mock_prepare_toolchain, mock_exists,
         self.assertEqual(args[1]['app_config'], None,
                          "prepare_toolchain was called with an incorrect app_config")
 
+    @patch('tools.build_api.intelhex_offset')
+    @patch('tools.config')
+    def test_merge_region_no_fit(self, mock_intelhex_offset, mock_config):
+        """
+        Test that merge region fails as expected when part size overflows region size.
+        """
+        max_addr = 87444 
+        # create a dummy hex file with above max_addr
+        mock_intelhex_offset.return_value = IntelHex({0:2, max_addr:0})
+
+        self.assertEqual(0, mock_intelhex_offset.return_value.minaddr())
+        self.assertEqual(max_addr, mock_intelhex_offset.return_value.maxaddr())
```

Comment:
These asserts don't test any tools code.

Reply:
since we use the IntelHex() library to generate this object, we can forego these asserts?

Reply:
Yes.